src/pipecat/processors/filters/stt_mute_filter.py

```diff
@@ -71,28 +72,36 @@ class STTMuteFilter(FrameProcessor):
     feature. When STT is muted, interruptions are automatically disabled.
 
     Args:
-        stt_service: Service handling speech-to-text functionality
         config: Configuration specifying muting strategies
+        stt_service: STT service instance (deprecated, will be removed in future version)
         **kwargs: Additional arguments passed to parent class
     """
 
-    def __init__(self, stt_service: STTService, config: STTMuteConfig, **kwargs):
+    def __init__(self, config: STTMuteConfig, stt_service: Optional[STTService] = None, **kwargs):
         super().__init__(**kwargs)
-        self._stt_service = stt_service
         self._config = config
+        if stt_service is not None:
+            warnings.warn(
```

Comment:
Move `import warnings` here so we can remove the whole block in the future and we don't forget removing the `import`.